[WIP] Added ioctl kprobe to USM

val06 opened the pull request:
### What does this PR do

Added erpc handler to communicate with dd-java-agent capturing the TLS traffic from Java applications

### Motivation

we capture TLS traffic in java applications using our dd-java-agent who in turn sends the captured data via a ioctl with special request code. SystemProbe should capture the data by setting a kprobe on the ioctl handler

### Additional Notes

Should be synced with this [PR](https://github.com/DataDog/datadog-agent/pull/14620) 

### Describe how to test/QA your changes

full e2e test should test injection of dd-java-agent and getting the tls traffic in systemprobe as part of the CI.

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

nplanel committed:
Attach an agent to a java process

Add java hotspot support to load an agent .jar or .so at runtime.

nplanel committed:
add Java TLS config support

DD_SYSTEM_PROBE_CONFIG_ENABLE_JAVA_TLS_SUPPORT
system_probe_config.enable_java_tls_support

nplanel committed:
add Java TLS config support

DD_SYSTEM_PROBE_CONFIG_ENABLE_JAVA_TLS_SUPPORT
system_probe_config.enable_java_tls_support

nplanel committed:
compile java with java7 jdk 1.7.0

nplanel committed:
USMAgent.jar

nplanel committed:
USMAgent.jar

nplanel committed:
agent-usm-7.43.0.jar

nplanel committed:
agent-usm-7.43.0.jar

nplanel committed:
loadCommand string assign

nplanel committed:
circleci-1189-with-java

nplanel committed:
ebpf java tls, subscribe process monitor

nplanel committed:
ebpf java tls, subscribe process monitor

nplanel committed:
hotspot close unix conn on error

nplanel committed:
typo Exists()

nplanel committed:
typo Exists()

nplanel committed:
addrandomIDdoc

nplanel committed:
addrandomIDdoc

nplanel committed:
add doc jattach_test PID namespace

nplanel committed:
attachChownAttachFile

nplanel committed:
add _ in gotls and javatls => go_tls java_tls

nplanel committed:
hotspot-doc

nplanel committed:
simplyfysocketExists

nplanel committed:
ftypo

nplanel committed:
ftypo

nplanel committed:
addcheckconfigdisabledbydefault

nplanel committed:
reusable-private-socketPath

nplanel committed:
freturnnilinsteademptyfunc

nplanel committed:
dstCloseifcopyfailed

nplanel committed:
attachPIDPath

nplanel committed:
fixupmissingerrocheckontouchattachpid

nplanel committed:
simplyfyAttachCommand

nplanel committed:
fcleanupjavaUSMAgentJarPath

nplanel committed:
fcleanupjavaUSMAgentJarPath

nplanel committed:
renameto-agent-usm.jar

nplanel committed:
renameto-agent-usm.jar

hush-hush committed:
Fix check conf directory

Durring the migration to component the hardcoded directory 'etc/confd'
for check configuration was removed.

hush-hush committed:
Fix shipping of 'version-history.json' and 'registry.json' in flares

When migrating to component the logic to include /opt/datadog-agent/run/
was handled as a file instead of a folder. This broke collecting
'version-history.json' and 'registry.json from it.

hush-hush committed:
Fix datadog.yaml file name in flare

hush-hush committed:
Force file permission to 644 within a flare

nplanel committed:
https-usm-doc

nplanel committed:
fhotspotdoc

nplanel committed:
fnonakedreturn

nplanel committed:
fdocnspid

nplanel committed:
trapStat_tcast

nplanel committed:
frenamelocalsrcAgent

nplanel committed:
fnsDstPath

nplanel committed:
docdstClose

nplanel committed:
fremovedstpathonerror

nplanel committed:
parseresponseerror

nplanel committed:
fixupdoc

nplanel committed:
attachJVMusealternativepathattachpath

nplanel committed:
fixupattachdoc

nplanel committed:
fdoc

nplanel committed:
renameAttachagentPath

nplanel committed:
fsimplify

nplanel committed:
docfsuidgid

nplanel committed:
docTestInject

nplanel committed:
fsimplifytestreadability

nplanel committed:
fdocsubprog

nplanel committed:
fdocsubprog

nplanel committed:
agentusmjarconst

nplanel committed:
agentusmjarconst

nplanel committed:
javasubprogramlog

nplanel committed:
javasubprogramlog

nplanel committed:
fixupcleanupexec

nplanel committed:
fixupcleanupexec

nplanel committed:
removetodorandom

nplanel committed:
removetodorandom

nplanel committed:
renameauthID

nplanel committed:
renameauthID

juliogreff committed:
[workloadmeta/kubelet] Parse image ID if name is a SHA256

We now try to parse the resolved image ID if the image in the pod's
container status is a SHA256. This seems to happen when pinning the
SHA256 in the container spec. This fixes an issue where `image:` filters
in DD_CONTAINER_INCLUDE/DD_CONTAINER_EXCLUDE would not be respected.

nplanel committed:
move enable_go_tls_support and java_tls to service_monitoring_config

nplanel committed:
move enable_go_tls_support and java_tls to service_monitoring_config

nplanel committed:
Update pkg/network/java/hotspot.go

Co-authored-by: Guy Arbitman <guy20495@gmail.com>

nplanel committed:
Update pkg/network/java/hotspot.go

Co-authored-by: Guy Arbitman <guy20495@gmail.com>

nplanel committed:
revert-usm-gotls-config

nplanel committed:
revert-usm-gotls-config

nplanel committed:
addinternaldoc

nplanel committed:
doccircleci

nplanel committed:
remove p==nil check

nplanel committed:
remove p==nil check

nplanel committed:
faster scan response

nplanel committed:
check if the socket exist before doing the hotspot handshake protocol

nplanel committed:
check if agent-usm.jar is accessible

nplanel committed:
check if agent-usm.jar is accessible

nplanel committed:
dont change gotls feature name

nplanel committed:
removemanifest and TestAgent code

nplanel committed:
removemanifest and TestAgent code

0intro committed:
[security-agent] remove redundant String() in compliance agent log (#15026)

KSerrania committed:
[invoke] Print summary of test failures at the end of inv test (#14682)

Updates the inv test command to print a summary of failed tests at the end of a run, across all modules and flavors that were tested, to more easily identify the list of failures, without having to visually parse the full job logs.

akarpz committed:
[system-probe][NET-2891] Fix tcp retransmit count (#14740)

* [NET-2891] initial pass at changes to prebuilt code

* [NET-2891] use retrans_out for runtime compiled tcp_retransmit counter

* [NET-2891] runtime compiled version of tcp_retrans updates

* [NET-2891] remove debug comment

* [NET-2891] fix log

* [NET-2891] update bytecode

* [NET-2891] code review comments, regenerate license

* [NET-2891] newline

* [NET-2891] fix probe definitions

* [NET-2891] update comment

* [NET-2891] runtime compilation fixes

* [NET-2891] fix byte padding for args init

* [NET-2891] fix formatting

* testing debug logic

* more debug logic, added some config for the map

* [NET-2891] enable kretprobe and remove debug

* [NET-2891] disable bpf debug be default

* [NET-2891] update bytecode

* [NET-2891] make function as maybe unused

* [NET-2891] handle different paths of incremental vs absolute retransmit counters

* [NET-2891] use enum to track increment vs absolute retransmits

* [NET-2891] change enum values

* [NET-2891] move retrans code to runtime tracer

* pulled in new gitignore

* Revert "pulled in new gitignore"

This reverts commit b4b0df587aeb6b6f655ea90d7bc96ae250934170.

* remove runtime gen files, code review comments

* [NET-2891] use retransmit count none in runtime tracer

* [NET-2891] use retransmit_count_none in handle_tcp_stats

* [NET-2891] nit comments from code review

* [NET-2891] try to get runtime compilation working on 4.4 kernel

akarpz committed:
[system-probe][NET-2891] Fix tcp retransmit count (#14740)

* [NET-2891] initial pass at changes to prebuilt code

* [NET-2891] use retrans_out for runtime compiled tcp_retransmit counter

* [NET-2891] runtime compiled version of tcp_retrans updates

* [NET-2891] remove debug comment

* [NET-2891] fix log

* [NET-2891] update bytecode

* [NET-2891] code review comments, regenerate license

* [NET-2891] newline

* [NET-2891] fix probe definitions

* [NET-2891] update comment

* [NET-2891] runtime compilation fixes

* [NET-2891] fix byte padding for args init

* [NET-2891] fix formatting

* testing debug logic

* more debug logic, added some config for the map

* [NET-2891] enable kretprobe and remove debug

* [NET-2891] disable bpf debug be default

* [NET-2891] update bytecode

* [NET-2891] make function as maybe unused

* [NET-2891] handle different paths of incremental vs absolute retransmit counters

* [NET-2891] use enum to track increment vs absolute retransmits

* [NET-2891] change enum values

* [NET-2891] move retrans code to runtime tracer

* pulled in new gitignore

* Revert "pulled in new gitignore"

This reverts commit b4b0df587aeb6b6f655ea90d7bc96ae250934170.

* remove runtime gen files, code review comments

* [NET-2891] use retransmit count none in runtime tracer

* [NET-2891] use retransmit_count_none in handle_tcp_stats

* [NET-2891] nit comments from code review

* [NET-2891] try to get runtime compilation working on 4.4 kernel

guyarb committed:
usm: upgraded pgdriver version to indirectly upgrade mellium.in/sasl version due to a CVE ofound (#15030)

* usm: upgraded pgdriver version to indirectly upgrade mellium.in/sasl version due to a CVE ofound

* Fixed go.sum

safchain committed:
[CWS] fix signal test (#15025)

nplanel committed:
javaslowstart

kkhor-datadog committed:
[process-agent] Support dynamically enabling profiling for process agent from CLI (#14995)

Adds support for dynamically enabling profiling for the process agent from the CLI

alexandre-normand committed:
pkg/obfuscate: Fix parsing of sqlserver identifiers enclosed in square brackets (#15019)

* DBM-2010 Fix parsing of sqlserver literals enclosed in square brackets

ajgajg1134 committed:
.gitlab: move APM benchmark job to manual only (#15036)

ian28223 committed:
fix datatype (#13791)

related to #13770

davidor committed:
[AD/prometheus] Ignore headless services (#15031)

julien-lebot committed:
Fix stop service (#15035)

hush-hush committed:
Use DMI on EC2 Nitro instances to get host aliases

The Agent now leverage DMI information on Unix to get the instance ID on AWS EC2 when the metadata endpoint fails or
is not accessible. The instance ID is exposed throught DMI only on AWS Nitro instances.

This will not change the hostname of the Agent upon upgrading but will add to the list of host aliases.

ahmed-mez committed:
auto instru: add rc provider (#15008)

katiehockman committed:
pkg/obfuscate: use github.com/outcaste-io/ristretto instead of github.com/dgraph-io/ristretto (#15005)

Migrate the usage of github.com/dgraph-io/ristretto to github.com/outcaste-io/ristretto

mx-psi committed:
pkg/trace/api: remove unused internal OTLP HTTP server (#14965)

* [pkg/trace/api] Remove unused OTLP HTTP server

* [pkg/trace] Remove protocol argument

* Remove unnecessary fmt.Sprintf

* Fix tests

paulcacheux committed:
[CWS] cleanup last uses of `jsonschema_description` (#15050)

nplanel committed:
wait java process to start the JVM to avoid crashing the java process by sending SIGQUIT too early on loaded platform

nplanel committed:
test log java -version

nplanel committed:
5sec

nplanel committed:
testdumpinfoonerror

nplanel committed:
18threadsmin

purple4reina committed:
[Serverless] Merge `serverless/main` to `main` (#14980)

* [Serverless] change account (#14755)

* Aj/buffer cold start span data (#14664)

* wip dirty commit - trace being created but not flushed properly. No further traces appearing

WIP: more debugging. StopChan properly set up

feat: Starting coldstart creator as a daemon, and recieving data from two channels. Todo: spec

feat: Update specs to write to channels

feat: Merge conflicts resolved for tests

feat: Use smaller methods to handle locking

fix: pass coldstartSpanId to sls-init main

feat: Remove default

feat: Use Millisecond as Second is far longer than necessary

feat: No need to export ColdStartSpanId

fix: update units

feat: Directionality for lambdaSpanChan as well as for initDurationChan

fix: No need for the nil check, I need to stop javascripting my go

feat: ints

* feat: rebase missing changes from merge commits

* feat: update ints after moving accounts

* Empty commit to trigger ci

* [Serverless] Fix flaky integration tests and make them more easily maintainable. (#14783)

* Retry serverless integration test failures automatically. (#14801)

* [Serverless] Allow some keys to be option in serverless integration tests. (#14827)

* Ability to remove items from the json.

* Remove items from snapshot.

* Do not expect spans when there is no spans object. (#14396)

* [Serverless] Improve stability of two tests. (#14895)

* Increase timeout while decreasing test time.

* Increase timeout in test.

* [Serverless] Consolidate log normalization to single file for integration tests. (#15004)

* Consolidate log normalization to single file.

* Save raw logs to a temp dir.

* Fix linting issues.

Co-authored-by: Maxime David <maxime.david@datadoghq.com>
Co-authored-by: AJ Stuyvenberg <astuyve@gmail.com>

derekwbrown committed:
Fixes multiple problems with http processing/tagging on Windows. (#15022)

* Fixes multiple problems with http processing/tagging on Windows.
- There was an offset error in which the port was not properly computed
  on ipv6 connections
- There was a problem with computing whether an ipv6 address was loopback or
  not
- The fullpath indication (which is used to compute the key) was not
  properly being computed.  This led to the same tuple being used
  as a different key, so transactions were not properly combined.

* fix grammar error in release notes

L3n41c committed:
Add the plumbing in the agent forwarder to submit container images and SBOM (#14962)

ogaca-dd committed:
Improve documentation for BundleParams (#15011)

ahmed-mez committed:
pkg/clusteragent/admission: add unit tests (#15044)

paulcacheux committed:
[CWS] bump syscall table + extract into separate task (#15061)

* 5.19 -> 6.1

* switch syscall table generator from go generate to task

* extract linux version

Commit:
[gitlab] Temporarily disable SUSE Agent 5 upgrade tests (#15055)

AlexandreYang committed:
[corechecks/snmp] Add LLDP remote device IP address (#14946)

nplanel committed:
6threadon1core

safchain committed:
[CWS] add discarders eBPF unit test (#14471)

* [CWS] add discarder retention ut

* add another test

* add a unit test task

* add trace param

* make eBPF test part of the CI

* fake time to speed up tests

* bump baloum version

* add more tests

modernplumbing committed:
[CWS Agent] Moving SecAgent subcommands to new dir part 2 (#14915)

* moving flare command to subcommands dir

* consolidating and moving secagent config package

* moving runtime to subcommands dir

* moved check subcommand, updated compliance subcommand which is the entry point to check funcs

* moving compliance cmd to subcommand dir

* exporting CliParams and RunCheck in Check subcommand for Compliance tests

* fixing cluster agent entry point into the check subcommand

L3n41c committed:
Add `container_image` core check (#14567)

L3n41c committed:
Add `container_image` core check (#14567)

julien-lebot committed:
Reorganize the specs for some kitchen test (#15027)

AlexandreYang committed:
[check command] Add `--instance-filter` option (#15034)

nplanel committed:
15mintests

clarkb7 committed:
Migrate systray to an fx.App (#14985)

Deprecate single-dash args and add double-dash args

Move code from cmd/systray to comp/systray

Update UAC manifest to requireAdministrator

Fix log file and add `system_tray.log_file` configuration option.

EmmaOnThursday committed:
epforwarder: update dbm samples endpoint prefix (#15053)

dbm-metrics-intake and dbquery-intake resolve to the same IPs. This change cleans up code so that we're only referencing one endpoint name.

nplanel committed:
seqtests

xornivore committed:
[process-agent] Refactor Check interface (#15063)

* [process-agent] Refactor Check interface

- Refactors Check interface to consolidate CheckWithRealTime features
- This will simplify integration with components in the future PRs since it eliminates casts

* Address feedback from @just-chillin

xornivore committed:
[process-agent] Refactor Check interface (#15063)

* [process-agent] Refactor Check interface

- Refactors Check interface to consolidate CheckWithRealTime features
- This will simplify integration with components in the future PRs since it eliminates casts

* Address feedback from @just-chillin

val06 committed:
added support for java tls ioctl

val06 committed:
added support for java tls ioctl

val06 committed:
added support for java tls ioctl

val06 committed:
added support for java tls ioctl

nplanel committed:
testlimtedby3min

nplanel committed:
checkjavaage5s

nplanel committed:
24mult5

val06 committed:
added support for java tls ioctl

val06 committed:
added support for java tls ioctl

val06 committed:
added support for java tls ioctl

val06 committed:
added support for java tls ioctl

val06 committed:
adding connection to the bpf_map when handling the request

val06 committed:
adding connection to the bpf_map when handling the request

val06 committed:
adding connection to the bpf_map when handling the request

paulcacheux committed:
include `functests` in `DD_PIPELINE_ID` for system probe and security agent functests (#15043)

* include `functests` in DD_PIPELINE_ID for system probe and security agent functests

* simpler/shorter pipeline_id

nplanel commented on pkg/network/ebpf/c/protocols/java-tls-erpc.h:
```diff
@@ -0,0 +1,86 @@
+
+    //read the actual length of the message (limited by HTTP_BUFFER_SIZE)
+    if (0 != bpf_probe_read_user(&bytes_read, sizeof(bytes_read), data)){
+        u64 pid_tgid = bpf_get_current_pid_tgid();
```
add the pids lines under `#ifdef DEBUG`
so you can remove the forced if dependency

Commit:
[install_script] Backport removal of RPM signing key 4172A230 (#15082)

AlexandreYang committed:
[corechecks/snmp] LLDP resolve local interface (#14991)

safchain committed:
[CWS] fix rule in error reported twice (#15084)

nplanel committed:
Add java package in our circle-ci image (#14665)

safchain committed:
[CWS] add inode to pid context to detect exec loss (#14661)

* [CWS] add revision to pid context

* use inode instead of revision

* Fix post rebase

L3n41c committed:
Fix serializer tests flakiness (#15093)

coignetp committed:
[RCM-632] Add UUID in request (#15088)

* Add org uuid field

* Add org uuid in request

* Remove generate file

* Comment exported method

wiyu committed:
fix the receiver name consistency (#15068)

ajgajg1134 committed:
Add limits to allocated dictionaries, prevent browser cross-site requests (#15067)

mx-psi committed:
pkg/trace/api: Move semantic conventions to separate internal package (#14963)

* [pkg/trace/api] Move semantic conventions to separate internal package

* Rename to shared

* Move tagContainersTags back to API package

* Rename package to 'header'

* Fix Windows build

L3n41c committed:
Factorize queue code duplicated at two places (#15098)

* Factorize the aggregating queue used by the SBOM and container image checks

* Mock time functions to make tests more reliable

val06 committed:
- added ioctl kprobe to ebpf_javatls subprogram

val06 committed:
- added ioctl kprobe to ebpf_javatls subprogram

val06 committed:
Merge remote-tracking branch 'origin/valeri.pliskin/java-tls-ioctl' into valeri.pliskin/java-tls-ioctl

# Conflicts:
#	pkg/network/ebpf/c/protocols/java-tls-erpc.h
#	pkg/network/ebpf/c/runtime/http.c

blt committed:
[single-machine-performance] Push agent containers to SMP ECR (#14438)

* [single-machine-performance] Push agent container to SMP ECR

This commit is an attempt to introduce pushing containers from Agent CI for
single-machine-performance's Regression Detector in our isolated
infrastructure. Much like we have done for vectordotdev/vector we intend to run
the Regression Detector on Agent changes, giving a reasonable statistical
guarantee that a change does or does not modify Agent performance by more than
random chance. In order for the Regression Detector to run jobs it must have
access to a 'baseline' and 'comparison' target. Baseline in this project would
be a container built from current `main` branch, comparison would be a container
built from the tip of a PR.

The main thing demonstrated here is that the team credentials SMP has created
for Agent are functional and are able to push up a containers, in a way that is
acceptable to Agent Platform. I have ammended `.docker_build_job_definition` to
mirror every created container to single-machine-performance's ECR, noting that
the tag now avoids the use of `CI_PIPELINE_ID`. In a later commit we will
introduce job submission and will rely on being able to compute the tag of a
previous pipeline's container from available Gitlab metadata, specificall
`CI_COMMIT_SHA` for the comparison container and whatever metadata maps to the
base branch's current SHA, `CI_MERGE_REQUEST_SOURCE_BRANCH_SHA`?

There are two outstanding questions regarding this work that I am aware of:

* Is there a race condition present between the triggering of this pipeline vs
main if users squash commits?
* Should we grant the exisitng CI user permissions into
single-machine-performance rather than use an issued bot account as done
presently and for vectordotdev/vector?

We've successfully demonstrated pushing up containers in a previous iteration of
this work, see https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/195939127.

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

* PR feedback

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

* trim ECR URL out of destination

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

* correct job dependency

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

* drop parallel.matrix

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

dependabot[bot] committed:
Bump github.com/Microsoft/hcsshim from 0.9.4 to 0.9.6 (#14785)

Bumps [github.com/Microsoft/hcsshim](https://github.com/Microsoft/hcsshim) from 0.9.4 to 0.9.6.
- [Release notes](https://github.com/Microsoft/hcsshim/releases)
- [Commits](https://github.com/Microsoft/hcsshim/compare/v0.9.4...v0.9.6)

---
updated-dependencies:
- dependency-name: github.com/Microsoft/hcsshim
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

derekwbrown committed:
[Windows] implement mapping of pid to service name (#15039)

* [Windows] implement mapping of pid to service name

Checks the pid against the table of SCM controlled processes.
If it's SCM controlled, returns the service information.

Because we must enumerate the entire SCM (there doesn't seem to be an api
for that), the SCMManager object maintains a cache of objects, and refreshes
only when it sees a PID it hasn't seen before.

On a machine with high process churn, this could still result in a lot
of accesses.  However, if process agent queries only when doing the
process check (i.e. every 30s), then it should only iterate the list
once per 30s.

* ci fixes, add tests

* fix test/improper conversion of data buffer

* review feedback

* more review feedback

* Rename structure

guyarb committed:
Update README.md (#15126)

* Update README.md

* Update README.md

* Update README.md

safchain committed:
[CWS] remove useless variable (#15112)

safchain committed:
[CWS][SEC-4478] add RC to e2e tests (#14877)

* [CWS] add RC to e2e tests

* fix host name

* check remote-config before file

* test embedded policy

* use a configmap to have a first policy

* make rc configurable

github-actions[bot] committed:
CWS: sync BTFhub constants (#15123)

Co-authored-by: paulcacheux <paulcacheux@users.noreply.github.com>

YoannGh committed:
[CWS] rename json fields to make them less misleading (#15097)

ahmed-mez committed:
pkg/trace/testutil: improve the randomization in test spans generator (#15108)

ruthnaebeck committed:
DOCS-2215 Add @env variables to datadog.yaml (#10069)

Co-authored-by: hestonhoffman <hestonhoffman@gmail.com>

paulcacheux committed:
[CSPM] e2e remote configuration fix (#15130)

* Add missing remote_configuration_enabled parameters in CSPM workflow

* add other missing parameters

spikat committed:
[CWS] chain only different binaries on activity dumps (#15095)

spikat committed:
[CWS][SEC-6381] Update template configuration file to add activity dumps and network detection parameters (#14835)

paulcacheux committed:
[CWS] remove unsafe usage in `ScopedVariables` (#15134)

gh123man committed:
Revert Remove CCA_IN_AD flag and related unused code  (#15115)

* Revert "Remove `CCA_IN_AD` flag and related unused code (#14955)"

This reverts commit 394ac59ad9707dfff8009c0dec03320d1df20098.

# Conflicts:
#	cmd/agent/common/autodiscovery.go

dependabot[bot] committed:
Bump github.com/hashicorp/consul/api from 1.13.0 to 1.15.3 (#13978)

Bumps [github.com/hashicorp/consul/api](https://github.com/hashicorp/consul) from 1.13.0 to 1.15.3.
- [Release notes](https://github.com/hashicorp/consul/releases)
- [Changelog](https://github.com/hashicorp/consul/blob/main/CHANGELOG.md)
- [Commits](https://github.com/hashicorp/consul/compare/v1.13.0...api/v1.15.3)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/consul/api
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Lénaïc Huard <lenaic.huard@datadoghq.com>

guyarb committed:
usm: classification: Fixed AMQP publish flakiness (#15071)

* usm: classification: Fixed AMQP publish flakiness

Publish has no response, so added classification by close-reply and close-ok messages.

* Moved closing of the client into the teardown

* Fixed test error

* Removed wrong bpf debug setup

clarkb7 committed:
Set windows-agent as CODEOWNER for systray (#15117)

gh123man committed:
Fix shutdown deadlock in docker socket tailer (#15138)

* fix shutdown deadlock in docker socket tailer

dependabot[bot] committed:
Bump github.com/CycloneDX/cyclonedx-go from 0.6.0 to 0.7.0 (#15081)

* Bump github.com/CycloneDX/cyclonedx-go from 0.6.0 to 0.7.0

Bumps [github.com/CycloneDX/cyclonedx-go](https://github.com/CycloneDX/cyclonedx-go) from 0.6.0 to 0.7.0.
- [Release notes](https://github.com/CycloneDX/cyclonedx-go/releases)
- [Changelog](https://github.com/CycloneDX/cyclonedx-go/blob/master/.goreleaser.yml)
- [Commits](https://github.com/CycloneDX/cyclonedx-go/compare/v0.6.0...v0.7.0)

---
updated-dependencies:
- dependency-name: github.com/CycloneDX/cyclonedx-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Adapt conversion functions to CycloneDX/cyclonedx-go 0.7.0

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Lénaïc Huard <lenaic.huard@datadoghq.com>

p-lambert committed:
[usm] Fix `ProcessMonitor` termination (#15140)

* [usm] Fix `ProcessMonitor` termination

* Ensure `refcount` doesn't go below zero

truthbk committed:
[gitlab] update builderimages for go 1.19.4 builds (#14930)

* [gitlab] update builderimages for go 1.19.4 builds

* [reno] adding relnote for golang 1.19.4 update

* [golang] bumping 1.19.4 in other relevant places

* [circleci] updating images

* [golangci-lint] fixing issues after 1.19.4 bump

* [gitlab] source_test: force external linker when running -race - troubleshooting

* [omnibus] devtoolset changes not required w/RHEL when building with centos 7+

* Revert "[gitlab] source_test: force external linker when running -race - troubleshooting"

This reverts commit 8a7d7344297554cf0eb42ee7f59ed847787ce217.

* [gitlab] source_test: use centos7 based images to enable -race detector

* [golangci-lint] address new format issue after merge

* Apply suggestions from code review

Co-authored-by: Vickenty Fesunov <vickenty@users.noreply.github.com>

* [golangci-lint] address missed lints on windows

* [gitlab] updating buildimages yet again

* [gitlab] setting buildimages after buildimages merged to main

Co-authored-by: Vickenty Fesunov <vickenty@users.noreply.github.com>

richl-dd committed:
[Windows] Create SCM and Windows Service utilities

Adds a set of SCM and Windows Service utility functions

akarpz committed:
Revert "[system-probe][NET-2891] Fix tcp retransmit count (#14740)" (#15141)

This reverts commit e677798efdfc27223f9c036c1fc2e429b2dd24e7.

xornivore committed:
[process-agent] Remove check singletons (#15121)

* [process-agent] Remove check singletons

* Address feedback from @just-chillin

guyarb committed:
Simplified the interface of httptx (#15146)

* Simplified the interface of httptx

* Fixed error

safchain committed:
[CWS] refactor eval options (#15132)

* [CWS] refactor eval options

* fix windows

val06 committed:
- added ioctl kprobe to UndefinedProbes list of javaTlsProgram

val06 committed:
- added ioctl kprobe to UndefinedProbes list of javaTlsProgram

safchain committed:
[CWS] enable ring buffer by default (#15111)

L3n41c committed:
Admission controller: support injecting multiple libs in the same pod (#14736)

FlorianVeaux committed:
[NDM] Do not send empty tags for snmp.interface.status (#15157)

* Do not send empty tags for snmp.interface.status

* fix test

BaptisteFoy committed:
[RCM-598] upgrade(remote-config): Use layered gRPC client between trace-agent & core-agent (#15100)

* upgrade(remote-config): Bump message size limit to 500MB

* fix(size): Size down to 110MB max

* refactor(auth): Refactor RC auth

* fix(interface): Remove opts

spikat committed:
[CWS] AD: drop event if its process lineage is incomplete and add a guard to avoid sending empty dumps (#15013)

In addition, two new metrics introduced to trace these new drops

davidor committed:
[tasks/licenses] Don't call `open` on dirs (#15161)


Co-authored-by: Alexandre Menasria <47357713+amenasria@users.noreply.github.com>

daniel-taf committed:
[process-agent] Set HintMask in CollectorProc during process checks. (#14759)

Adds a new process discovery hint in the process agent when the regular process and container checks run.

pducolin committed:
[fake-datadog] fix index list numbers (#15154)

kangyili committed:
Enable orchestrator manifest collection by default (#15094)

manifest collection GA

KSerrania committed:
Update CODEOWNERS and JOBOWNERS after team and job renames (#15021)

Updates the JOBOWNERS and the GITHUB_SLACK_MAP file to account for recent team and job changes.

guyarb committed:
Changed TSM to USM (#15143)

nplanel committed:
Network USM : add java TLS support (#14620)

Adding support to attach a live java process and send it "agent-usm.jar" runtime agent payload

Supporting JVMTI Hotspot mechanism

Configuration:
DD_SERVICE_MONITORING_CONFIG_ENABLE_JAVA_TLS_SUPPORT = true
service_monitoring_config:
  enable_java_tls_support: true


Co-authored-by: Guy Arbitman <guy20495@gmail.com>

AlexandreYang committed:
Fix instance-filter error type (#15163)

val06 committed:
- added ioctl probe activation
- fixed registers reading in do_vfs_ioctl kprobe

val06 committed:
Merge remote-tracking branch 'origin/valeri.pliskin/java-tls-ioctl' into valeri.pliskin/java-tls-ioctl

# Conflicts:
#	.circleci/config.yml
#	pkg/network/ebpf/c/protocols/java-tls-erpc.h
#	pkg/network/ebpf/c/runtime/http.c
#	pkg/network/java/hotspot.go
#	pkg/network/java/jattach.go
#	pkg/network/java/jattach_test.go
#	pkg/network/protocols/http/ebpf_javatls.go
